### DISCUSSION TITLE
Relocate Media folder to be a shared folder

### DISCUSSION THREAD

**Discussion by minhdn2** [OrchardCMS/OrchardCore#16118] [Q&A]
Hello,

Currently I need to deploy Orchard Core headless on 2 servers for load balancing, so the images need to be the same.

I'm planning to make the **Media** folder to become a shared folder so that both site on different servers can access but not sure how to do this. Could you guys please help to advise on how can I do this?

If there is any other ways please help to advise also.

Many thanks in advance.

**Comment by deanmarcussen** [comment]
Generally we would recommend you use the `OrchardCore.Media.Azure` feature for this, to store your media on blob storage.

Alternatively (and it's hard to advise exactly, without knowing your server environment/hosting) you can relocate the location of the `App_Data` to a shared volume.

See docs here https://docs.orchardcore.net/en/dev/docs/reference/core/Configuration/#orchard_app_data-environment-variable

**Comment by minhdn2** [comment]
Hi @deanmarcussen 

Since the shared folder is located in AWS so I think `OrchardCore.Media.Azure` feature is not relevant for my situation.

Beside that, the 2 sites will be hosted on IIS.

I've been looking for how to modify the config file or make changes in code but not yet found any working solution for the latest Orchard Core version.

Not sure if the doc is relevant for my situation since what I need is the **Media** or the **App_Data** folder to be on different server from the one that hosting the CMS - not **Tenants** related.

Please help to check and advise,
Many thanks.

**Comment by deanmarcussen** [comment]
You might consider implementing an S3 provider (that would be the recommendation)

But if you just want a shared volume in AWS, normally, you would mount the volume in Windows, and then you can use the `Orchard_App_Data` to map the `App_Data` folder to the mounted volume.

The same applies to the `Media` path, you can move that out of the `App_Data` folder to a `root folder / drive volume` by configuring the `AssetsPath` 

This is documented here https://docs.orchardcore.net/en/dev/docs/reference/modules/Media/#configuration

**Comment by minhdn2** [comment]
Hi @deanmarcussen ,

Does `AssetsPath `support absolute path? If I keep on going back from the current Media folder, only then I can relocate the Media folder.

However, if I insert the absolute path to the `AssetsPath`, the website cannot start up and run correctly.

Here is my setup, the commented one is not working.
```
"OrchardCore": {
	"OrchardCore_Media": {
		"AssetsPath": "../../../Media",
		//"AssetsPath": "C:\Users\dnminh\Documents\My Web Sites\cms-local\shared_folder\Media ",
	}
  }
```

Please help to check and advise,
Many thanks.

**Comment by deanmarcussen** [comment]
You can go absolute with a leading `/` character.

So maybe try with a unc style path

**Comment by minhdn2** [comment]
hi @deanmarcussen ,

I've been trying with:
`C:/shared_folder`
`C://shared_folder`
`C:\shared_folder`
`C:\\shared_folder`
but none of them works.
![image](https://user-images.githubusercontent.com/35803003/81944467-016c7f00-9627-11ea-8201-779393e5cc38.png)


Please help to check and advise,
Many thanks in advance

**Comment by deanmarcussen** [comment]
try `/shared_folder`, i.e. leading absolute path.

**Comment by minhdn2** [comment]
hi @deanmarcussen ,

`/shared_folder'` didn't work, the site cannot start.
I tried with `\\shared_folder` but it only relocate to the root of the current drive (my website is stored in `C:\...\...\...\my_website` and the shared_ folder is created in `C:\shared_folder`)

Could you please advise on how can I relocate to a shared folder in other drive (i.e: `D:\shared_folder`) or a shared folder in other computer (i.e: `\\10.x.x.x\shared_folder`) ?

Many thanks in advance.

**Comment by deanmarcussen** [comment]
Let me more specific.

The check on the folder paths to get an absolute path is if the leading character contains either `/` or `\` 

There will be no check for anything like `C:` - that is not platform independant (i.e. would not work for linux), so do not try to do that.

as I said before if you want to locate to network driver use a unc share path.

if you want to move it for development perhaps try creating a symbolic link
https://www.techrepublic.com/article/how-to-take-advantage-of-symbolic-links-in-window-10/
